ocaml/src/proxy_server.ml

```diff
@@ -596,6 +607,27 @@ let proxy_protocol (alba_client : Alba_client.alba_client)
        fun stats () -> alba_client # get_alba_id
     | HasLocalFragmentCache ->
        fun stats () -> alba_client # has_local_fragment_cache |> Lwt.return
+    | UpdateSession ->
+       fun stats args ->
+       begin
+         let () = List.iter
+           (function
+            | "manifest_ser", vo ->
+               begin
+                 match vo with
+                 | None -> ()
+                 | Some v ->
+                    let new_version = deserialize Llio.int8_from v in
+                    ProxySession.set_manifest_ser session new_version
+               end
+            | key,_ ->
+               let open Nsm_model in
+               Err.failwith Err.Session_update_error ~payload:key
```

Comment:
So for unknown keys this throws an exception...
That makes it harder for future clients that want to specify multiple keys (in 1 call), as they might talk to an old server which only supports a subset of the keys. So they would have to fall back to multiple UpdateSession calls anyway (then why take a list in the first place).
Maybe it makes more sense here to return the keys that the server was able to interpret?

(Similar remark for nsm host session keys)